### ISSUE TITLE
Add description of 'metadata' checkbox to help

### ISSUE THREAD

**Issue by njh** [njh/node-red-contrib-osc#8]
It is currently unclear what the purpose/impact for checking the of the 'Metadata' box.

The `osc.js` README states:

> Type-annotated argument objects contain two properties: type, which contains the OSC type tag character (e.g. "i", "f", "t", etc.) and the raw value.

```
{
    type: "f", // OSC type tag string
    value: 444.4
}
```

> If you are using type-annotated arguments, you should also set the metadata option to true when you instantiate your OSCPort instance (or in the options argument to osc.writeMessage if you're using the low-level API).

**Comment by njh** [comment]
@natcl are you able to look at this please?

**Comment by natcl** [comment]
I'm not sure I understand the issue.

The purpose (from what I understand) is to know the exact type that was sent.  So if you receive a number, you could differentiate wether it's a float  or an int (since everything is a float in javascript)

For sending values the checkbox doesn't have any impact as you can choose to send a raw payload which the underlying will try to set the correct type, or if you want to force the sending of a specific type (for example to a software that would respond only to ints) then you send it an object:

```
{
    type: "i", // OSC type tag string
    value: 444
}
```

Or perhaps I don't understand the message clearly ?

Perhaps what you mean is that we could always ask the library to send the metadata and enclose the type in a msg.type property ?  This would allow to remove the checkbox from the gui.

**Comment by natcl** [comment]
Or, perhaps we could do something like this:

Add a msg.osc.raw object that contains the full osc message while the normal topic/payload work as they do now with the checkbox unchecked ?

Something like this: 

``` javascript
msg.osc.raw = osc.readPacket(msg.payload, {"metadata": true, "unpackSingleArgs": false});
msg.payload = osc.readPacket(msg.payload, {"metadata": false, "unpackSingleArgs": true});
```

**Comment by njh** [comment]
Sorry, this isn't about changing the code/the way it works.

It is just about adding some help text to the UI, so it makes it clear to the user what pressing the checkbox does. I was confused about what it did, or why you would tick it - so thought other would be too.

**Comment by natcl** [comment]
Ah got you, I'll add some info to the help section!
Le lun. 10 oct. 2016 à 14:37, Nicholas Humfrey notifications@github.com a
écrit :

**Comment by natcl** [comment]
Just sent PR #11 let me know if it's ok !

**Comment by njh** [comment]
Fixed in #11
